Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/GazeContingent/FixWindowBufferedSamples/EyeLink_FixWindowBufferedSamples.m

```diff
@@ -0,0 +1,538 @@
+function EyeLink_FixWindowBufferedSamples(screenNumber)
+% EyeLink gaze-contingent demo that shows how to retrieve online gaze samples from a buffer.
+% In each trial central crosshairs are shown until gaze is detected continuously within a central 
+% square window for 500ms or until the space bar is pressed. An image is 
+% then presented until the space bar is pressed to end the trial.
+%
+% Usage:
+% Eyelink_FixWindowBufferedSamples(screenNumber)
+%
+% screenNumber is an optional parameter which can be used to pass a specific value to Screen('OpenWindow', ...)
+% If screenNumber is not specified, or if isempty(screenNumber) then the default:
+% screenNumber = max(Screen('Screens'));
+% will be used.
+%
+% This demo uses the 'GetNextDataType'/'GetFloatData' function pair that allows access to the following buffered samples and events
+% (See EyeLink Programmers Guide manual > Data Structures > FEVENT):
+%
+% STARTBLINK 3 (the start of a blink)
+% ENDBLINK 4 (the end of a blink)
+% STARTSACC 5 (the start of a saccade)
+% ENDSACC 6 (the end of a saccade)
+% STARTFIX 7 (the start of a fixation)
+% ENDFIX 8 (the end of a fixation)
+% FIXUPDATE 9 (a fixation update during a fixation)
+% SAMPLE_TYPE 200 (a sample)
+% MISSING_DATA -32768 (missing data)
+%
+% Use buffered data if you need to:
+% a) grab every single consecutive sample online
+% b) grab event data (e.g. fixation/saccade/blink events) online
+

```

Comment:
This will cut off all remaining help text -> You must use % for empty lines.

Reply:
a9106063a6389f731883f46a924e975472be507b